### PR TITLE
Feature/30 spawner schedule skeleton

### LINKED ISSUE #30: Spawner and Schedule Skeleton
## User stories
As a developer I want the skeleton of the classes for Spawner and Schedule in order to implement the Spawner MVP.

## Dependencies
Depends on: 
 - nothing

## (Wiki) References
- [Architecture Diagram](https://files.slack.com/files-pri/T048R6TD7MJ-F04T38B4KLM/download/architecture_v1.pdf?origin_team=T048R6TD7MJ) 

## Acceptance Criteria
- [x] Setup directory structure
    - [x] src/spawner
    - [x] src/schedule
    - [x] tests/spawner
    - [x] tests/schedule
- [ ] Setup abstract Spawner Class
- [ ] Setup Train Spawner Class
- [ ] Setup abstract Schedule Class
- [ ] Setup Train Schedule Class
- [ ] Setup abstract Base Strategy
- [ ] Setup Regular Startegy

### PR DESCRIPTION
**Pull request by CR1337**
Fixes #30

Skeleton is now ready and can be filled with implementation.

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [ ] Docs (code, wiki & diagrams) updated
- [ ] Breaking changes anounced
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: src/schedule/schedule_strategy.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+
+
+class ScheduleStrategy(ABC):
+    """An abstract ScheduleStrategy.
+    ScheduleStartegies implement the decision logic of
+    how a schedule determines whether a vehicle shoudl be spawned
```
Comment (instant-sky) [inline src/schedule/schedule_strategy.py]: little typo: "shoudl"

**Review by instant-sky — APPROVED** [review]
good work

File: src/schedule/schedule_strategy.py (hunk 1 above)
Comment (CR1337) [reply src/schedule/schedule_strategy.py]: is fixed

**Review by instant-sky — APPROVED** [review]
amazing improvement! 😯

**Pull request merged**